src/acb_theta/ctx_exp_inv.c

```diff
@@ -0,0 +1,35 @@
+/*
+    Copyright (C) 2024 Jean Kieffer
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+*/
+
+#include "acb.h"
+#include "acb_theta.h"
+
+void acb_theta_ctx_exp_inv(acb_t exp_inv, const acb_t exp, const acb_t z,
+    int z_is_real, slong prec)
+{
+    if (z_is_real)
+    {
+        acb_conj(exp_inv, exp);
+    }
+    else if (acb_contains_zero(exp)
+        || acb_rel_error_bits(exp) <= 0.75 * prec)
+    {
+        acb_t x;
+        acb_init(x);
+        acb_neg(x, z);
+        acb_exp_pi_i(exp_inv, x, prec);
+        acb_clear(x);
+    }
+    else
+    {
+        acb_inv(exp_inv, exp, prec);
+    }
```

Comment:
Is the logic in this file correct, considering that the presumably faster acb_inv branch is not reached by tests?

Reply:
Thanks for catching this ! I guess I was confused about what `acb_rel_error_bits` does. On 1 +- 0.25, is it supposed to return -2 right?
Then the correct line should probably be ` || acb_rel_error_bits(exp) >= -0.75*prec`. The goal was to only trigger recomputing an exponential when `acb_inv` would cause NaNs or large errors to appear.

Reply:
I think prefer to use ``acb_rel_accuracy_bits`` here. So ``acb_rel_accuracy_bits(exp) <= 0.75 * prec`` (or ``0.25 * prec``? it's not entirely clear what cutoff was intended). Would be nice to have this function tested anyway.

Reply:
Solved by https://github.com/flintlib/flint/pull/2182/commits/4253f1d8f3fafa5be7968e2ae5ebb9f380faa419